02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+    * [`u64`:`dust_limit_satoshis`]
+    * [`u64`:`max_htlc_value_in_flight_msat`]
+    * [`u64`:`htlc_minimum_msat`]
+    * [`u32`:`minimum_depth`]
+    * [`u16`:`to_self_delay`]
+    * [`u16`:`max_accepted_htlcs`]
+    * [`point`:`funding_pubkey`]
+    * [`point`:`revocation_basepoint`]
+    * [`point`:`payment_basepoint`]
+    * [`point`:`delayed_payment_basepoint`]
+    * [`point`:`htlc_basepoint`]
+    * [`point`:`first_per_commitment_point`]
+    * [`accept_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `accept_tlvs`
+2. types:
+   1. type: 1 (`option_upfront_shutdown_script`)
+   2. data:
+       * [`u16`:`shutdown_len`]
+       * [`shutdown_len*byte`:`shutdown_scriptpubkey`]
+
+
+Rationale and Requirements are the same as listed above,
+for [`accept_channel`](#the-accept_channel-message) with the following
+additions.
+
+#### Requirements:
+
+The accepting node:
+    - MAY respond with a `funding_satoshis` value of zero.
+
+#### Rationale
+
+The `funding_satoshis` is the amount of bitcoin in satoshis
+the *accepter* will be contributing to the channel's funding transaction.
+
+Note that `accept_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
+
+`channel_id` for the `accept_channel2` MUST be derived using a zero-d out
+basepoint for the peer's revocation basepoint.  This allows the peer to
+return channel-assignable errors before the *accepter*'s revocation
+basepoint is known.
+
+### Funding Composition
+Funding composition for channel establishment v2 makes use of the
+[Interactive Transaction Construction](#interactive-transaction-construction)
+protocol, with the following additional caveats.
+
+
+#### The `tx_add_input` Message
+
+No additional caveats or requirements.
+
+
+#### The `tx_add_output` Message
+##### Requirements
+The sending node:
+  - if is the *opener*:
+    - MUST send at least one `tx_add_output`,  which contains the
+      channel's funding output
+
+##### Rationale
+The channel funding output must be added by the *opener*, who pays its fees.
+
+
+#### The `tx_complete` Message
+
+Upon receipt of consecutive `tx_complete`s, the receiving node:
+  - if is the *accepter*:
+    - MUST fail the negotiation if:
+      - no funding output was received
+      - the value of the funding output is not equal to the sum of
+        `open_channel2`.`funding_satoshis` and `accept_channel2`.
+        `funding_satoshis`
+      - the value of the funding output is less than the `dust_limit`
+ - if is an RBF attempt:
+    - MUST fail the negotiation if:
+      - the transaction's total fees is less than the last
+        successfully negotiated transaction's fees
+      - the transaction does not share a common input with all previous
+        funding transactions
```

Comment:
I think we need to be more restrictive than that, we also cannot add new unconfirmed inputs (BIP 125 rule 2).
Or maybe should we say "the transaction follows the usual BIP 125 replacement rules" or something like it?

Reply:
Ideally yes, we'd specify that they can't break any of the BIP125 rules, including the "no new unconfirmed inputs". However, we can't really verify that without attempting to broadcast it to our local mempool. We *can* however verify that they're executing a doublespend (and in fact this is more important to verify than "no new unconfirmed inputs"), as failing to adhere to it might lead to *two* channel funding txs being included in the chain.

I tried to adhere to everything we absolutely needed to verify/confirm and leave out the things that we can't really (but if we could we should), such as the "no new unconfirmed inputs" rule. 

An RBF might fail to broadcast for a few reasons, and it's expected that they will.

Reply:
I could add a note that the *sender* shouldn't send any new unconfirrmed inputs, but there's no enforcement on the receiver side really.

Reply:
Note that if you're using `bitcoind`, you actually can perform the checks without broadcasting with the `testmempoolaccept` RPC (that's probably what I'd use when implementing).

> I could add a note that the sender shouldn't send any new unconfirrmed inputs, but there's no enforcement on the receiver side really. What do you think?

That sounds good enough!

Reply:
That's true.